consensus/src/common.rs

```diff
@@ -30,7 +30,48 @@ where
     fn state_hash(&self) -> Option<StateHash>;
 }
 
-impl Chain<ProtocolState> for ProtocolStateChain {
+impl Common<ProtocolState> for ProtocolState {
```

Comment:
This should be reverted back. Can you explain why we had to re-implement this. These methods are common across all ProtocolState instances.

Reply:
This is from @noot's legacy code as the following:
![Screenshot (285)](https://user-images.githubusercontent.com/8589224/142221228-f86a9f15-6a9e-42be-b04c-cce90d22efba.png)
![Screenshot (286)](https://user-images.githubusercontent.com/8589224/142221438-33a60075-717e-4d90-8155-6940783a1928.png)
We have both ProtocolStateChain and ProtocolState.

Reply:
This part is from @noot's legacy code
![Screenshot (285)](https://user-images.githubusercontent.com/8589224/142222377-e475c324-13c5-453c-82e8-762afd734201.png)
![Screenshot (286)](https://user-images.githubusercontent.com/8589224/142222408-9ab58201-7249-4090-bdb2-75b27478df5f.png)
:

Reply:
As in https://github.com/ChainSafe/mina-rs/pull/34/files#diff-09b1b8c7ec19c8817ec99388da62c7d423b51a868df0ec116ef9d4dde9919286R21 the `Chain` trait should be left as it is. `ProtocolStateChain` only implements that. We would just need to update the methods within that trait to match the new spec, if there are any. If you see this section: https://github.com/MinaProtocol/mina/blob/32f529d8d8d712a44ee75be66061ce08cbdc8924/docs/specs/consensus/README.md#51-common those are all the methods the Chain trait implements. Also we don't need to strictly abide by the exact names. The spec naming convention is just for readability purposes. We can rename variables and type names as it makes sense in our code. So that's why we don't have it as `Common` but rather as `Chain`

Reply:
Got it! Thank you for clarifying, Rahul.